lib/src/coap_client.dart

```diff
@@ -35,338 +55,353 @@ class CoapClient {
   /// If it is specified it will be prepared and used.
   /// Note that the host name part of the URI can be a name or an IP address,
   /// in which case it is not resolved.
-  CoapClient(this.uri, this._config, [this.request]);
-
-  final CoapILogger? _log = CoapLogManager().logger;
-  static final Iterable<CoapWebLink> _emptyLinks = <CoapWebLink>[
-    CoapWebLink('')
-  ];
-
-  /// The URI
-  Uri uri;
-  final DefaultCoapConfig _config;
-
-  String get _namespace =>
-      _config.poolUdpConnectionsByClient ? hashCode.toString() : '';
-
-  /// The endpoint. Once set, on the first request this endpoint is used
-  /// throughout this client for all subsequent requests.
-  CoapIEndPoint? endpoint;
-
-  int _type = CoapMessageType.con;
-  int _blockwise = 0;
-
-  /// The current request being processed by the client.
-  CoapRequest? request;
-
-  /// Timeout
-  int timeout = 32767; //ms
+  CoapClient(this.uri, this._config,
+      {this.addressType = InternetAddressType.IPv4, Duration? timeout}) {
+    this.timeout = timeout ?? Duration(milliseconds: _config.defaultTimeout);
+    _namespace = hashCode.toString();
+  }
```

Comment:
Yes, I don't know how we should treat the config. I figured it could make more sense to move our non-RFC timeout default (which we had apparently set to 32767 ms) there as well, so we have a single config. Then again, we already have an implicit total timeout specified in the config, so should we define yet another timeout for the client like we do now? Regardless, the config will likely span multiple RFC's (7252 for UDP, 8232 for TCP, 7641 for observe, 7959 for blockwise), so it will probably not map 1:1 to RFC 7252.

Reply:
Fixed the constructor: https://github.com/shamblett/coap/pull/76/commits/23f12832be6d414080a70c4193c1a4afe34c44e9

If you want I can remove `defaultTimeout` from the config and add a separate issue to discuss whether we should specify the timeout twice?

Reply:
> Fixed the constructor: [23f1283](https://github.com/shamblett/coap/commit/23f12832be6d414080a70c4193c1a4afe34c44e9)
> 
> If you want I can remove `defaultTimeout` from the config and add a separate issue to discuss whether we should specify the timeout twice?

That sounds like a good idea :)

Reply:
Fixed: https://github.com/shamblett/coap/pull/76/commits/175151b7968921eae2d53806b700e69a345a19d5